tests/sentry/api/test_event_search.py

```diff
@@ -768,271 +768,801 @@ def test_empty_string(self):
-        self.term1 = SearchFilter(
-            key=SearchKey(name="user.email"),
-            operator="=",
-            value=SearchValue(raw_value="foo@example.com"),
-        )
-        self.term2 = SearchFilter(
-            key=SearchKey(name="user.email"),
-            operator="=",
-            value=SearchValue(raw_value="bar@example.com"),
-        )
-        self.term3 = SearchFilter(
-            key=SearchKey(name="user.email"),
-            operator="=",
-            value=SearchValue(raw_value="foobar@example.com"),
-        )
-        self.term4 = SearchFilter(
-            key=SearchKey(name="user.email"),
-            operator="=",
-            value=SearchValue(raw_value="hello@example.com"),
-        )
-        self.term5 = SearchFilter(
-            key=SearchKey(name="user.email"),
-            operator="=",
-            value=SearchValue(raw_value="hi@example.com"),
-        )
+        self.term1 = ["equals", ["user.email", "foo@example.com"]]
+        self.term2 = ["equals", ["user.email", "bar@example.com"]]
+        self.term3 = ["equals", ["user.email", "foobar@example.com"]]
+        self.term4 = ["equals", ["user.email", "hello@example.com"]]
+        self.term5 = ["equals", ["user.email", "hi@example.com"]]
 
     def test_simple(self):
-        assert parse_search_query("user.email:foo@example.com OR user.email:bar@example.com") == [
-            SearchBoolean(left_term=self.term1, operator="OR", right_term=self.term2)
-        ]
+        result = get_filter("user.email:foo@example.com OR user.email:bar@example.com")
+        print("RES", result.conditions)
+        assert result.conditions == [[["or", [self.term1, self.term2]], "=", 1]]
 
-        assert parse_search_query("user.email:foo@example.com AND user.email:bar@example.com") == [
-            SearchBoolean(left_term=self.term1, operator="AND", right_term=self.term2)
-        ]
+        result = get_filter("user.email:foo@example.com AND user.email:bar@example.com")
+        assert result.conditions == [[["and", [self.term1, self.term2]], "=", 1]]
 
     def test_single_term(self):
-        assert parse_search_query("user.email:foo@example.com") == [self.term1]
+        result = get_filter("user.email:foo@example.com")
+        assert result.conditions == [["user.email", "=", "foo@example.com"]]
 
     def test_order_of_operations(self):
-        assert parse_search_query(
+        result = get_filter(
             "user.email:foo@example.com OR user.email:bar@example.com AND user.email:foobar@example.com"
-        ) == [
-            SearchBoolean(
-                left_term=self.term1,
-                operator="OR",
-                right_term=SearchBoolean(
-                    left_term=self.term2, operator="AND", right_term=self.term3
-                ),
-            )
+        )
+        print("RESULT", result.conditions)
+        assert result.conditions == [
+            [["or", [self.term1, ["and", [self.term2, self.term3]]]], "=", 1]
```

Comment:
As an example to turning the emails into variables, I assume this is correct but without looking above I'm not 100% sure that `foobar@example.com` is term1.

Reply:
Maybe the solution is to rename these to `self.foo`, `self.bar` etc to make it more obvious what the expected case is.

Reply:
I agree that would help.